invenio_circulation/transitions/transitions.py

```diff
@@ -38,6 +38,35 @@ def _ensure_valid_loan_duration(loan):
         raise TransitionConstraintsViolation(msg=msg)
 
 
+def _ensure_valid_extension(loan):
+    """Validate end dates for a extended loan."""
+    get_extension_max_count = current_app.config['CIRCULATION_POLICIES'][
+        'extension']['max_count']
+    extension_max_count = get_extension_max_count(loan)
```

Comment:
It looks like there is a mix of concerns, `extension_max_count` is a policy and `get_extention_max_count` takes loan as a parameter. If loan has anything to do with it extention_max_count could be a property of it.

Reply:
It is the same as the **checkout** policies. It enables the calculation of future complex policies.

Reply:
Do we want this policy to be overriden or extendable?

Reply:
@zzacharo extendable for sure! For example, `extension_max_count` will probably depends on patron type or item type.